src/tools/yarprobotstatepublisher/src/robotstatepublisher.cpp

```diff
@@ -78,6 +79,7 @@ bool YARPRobotStatePublisherModule::configure(ResourceFinder &rf)
 
     string modelFileName=rf.check("model",Value("model.urdf")).asString();
     m_period=rf.check("period",Value(0.010)).asDouble();
+    m_treeType=rf.check("tree-type", Value("SHALLOW")).asString();
```

Comment:
Referring to https://github.com/robotology/idyntree/pull/724#discussion_r471342972, maybe checking here that the value can be either SHALLOW or DEEP and return false already in the `configure` step would be even better.

Reply:
I agree